tests/bgp/test_bgp_stress_link_flap.py

```diff
@@ -151,84 +149,75 @@ def flap_fanout_interface_all(interface_list, fanouthosts, duthost, stop_event,
     fanout_flap_count = 0
 
     start_time = time.time()  # Record the start time
-    while not stop_event.is_set() and time.time() - start_time < test_run_duration:
+    while not stop_tasks and time.time() - start_time < test_run_duration:
         for port in interface_list:
-            if stop_threads:
+            if stop_tasks:
                 break
 
             fanout, fanout_port = fanout_switch_port_lookup(fanouthosts, duthost.hostname, port)
             if fanout and fanout_port:
                 logger.info("flap interface fanout {} port {}".format(fanout, fanout_port))
                 fanout.shutdown(fanout_port)
-                time.sleep(sleep_duration)
+                await asyncio.sleep(sleep_duration)
                 fanout.no_shutdown(fanout_port)
-                time.sleep(sleep_duration)
+                await asyncio.sleep(sleep_duration)
             else:
                 logger.warning("fanout not found for {} port {}".format(duthost.hostname, port))
 
         fanout_flap_count += 1
-        if stop_threads:
+        if stop_tasks:
             logger.info("Stop flap thread, breaking flap fanout {} dut {} interface {} flap count  {}".format(
                 fanouthosts, duthost, port, fanout_flap_count))
             break
 
 
-def flap_neighbor_interface(neighbor, neighbor_port, stop_event, sleep_duration, test_run_duration):
+async def flap_neighbor_interface(neighbor, neighbor_port, sleep_duration, test_run_duration):
     logger.info("flap neighbor {} interface {}".format(neighbor, neighbor_port))
     global neighbor_flap_count
     neighbor_flap_count = 0
 
     start_time = time.time()  # Record the start time
-    while not stop_event.is_set() and time.time() - start_time < test_run_duration:
+    while not stop_tasks and time.time() - start_time < test_run_duration:
         neighbor.shutdown(neighbor_port)
-        time.sleep(sleep_duration)
+        await asyncio.sleep(sleep_duration)
         neighbor.no_shutdown(neighbor_port)
-        time.sleep(sleep_duration)
-        if stop_threads:
+        await asyncio.sleep(sleep_duration)
+        if stop_tasks:
             logger.info("Stop flap thread, breaking flap neighbor {} interface {} flap count {}".format(
                 neighbor, neighbor_port, neighbor_flap_count))
             break
         neighbor_flap_count += 1
 
 
 @pytest.mark.parametrize("test_type", ["dut", "fanout", "neighbor", "all"])
-def test_bgp_stress_link_flap(duthosts, rand_one_dut_hostname, setup, nbrhosts, fanouthosts, test_type,
-                              get_function_completeness_level):
-    global stop_threads
+def test_bgp_stress_link_flap(duthosts, rand_one_dut_hostname, setup, nbrhosts, fanouthosts, test_type):
+    global stop_tasks
     global dut_flap_count
     global fanout_flap_count
     global neighbor_flap_count
-    global TEST_RUN_DURATION
 
     duthost = duthosts[rand_one_dut_hostname]
 
-    normalized_level = get_function_completeness_level
```

Comment:
Seems get_function_completeness_level has been removed? may you share the reason, thanks

Reply:
This was accidentally removed, added back.